src/sdb.c

```diff
@@ -449,7 +449,7 @@ static int sdb_set_internal (Sdb* s, const char *key, char *val, int owned, ut32
 			if (cas && kv->cas != cas) {
 				return 0;
 			}
-			if (vlen == kv->value_len && !strcmp (kv->value, val)) {
+			if (vlen == kv->value_len && strcmp (kv->value, val) != 0) {
```

Comment:
but yep. i have noticed that the test is not passing and for some reason is not failing on travis. thats why i didnt noticed that :(


> On 13 Sep 2016, at 16:03, Tiger Toes <notifications@github.com> wrote:
> 
> In src/sdb.c <https://github.com/radare/sdb/pull/109#discussion_r78564784>:
> 
> > @@ -449,7 +449,7 @@ static int sdb_set_internal (Sdb* s, const char *key, char *val, int owned, ut32
> >  			if (cas && kv->cas != cas) {
> >  				return 0;
> >  			}
> > -			if (vlen == kv->value_len && !strcmp (kv->value, val)) {
> > +			if (vlen == kv->value_len && strcmp (kv->value, val) != 0) {
> Thanks for your feedback, would this mean that the tests in cas.c (which appear to have also failed in the last successful build on the CI <https://travis-ci.org/radare/sdb/builds/152275982#L279>) are incorrect, not the code itself?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub <https://github.com/radare/sdb/pull/109/files/84236ae4023560bfbd2c1b9f0e1c32691dba2f19#r78564784>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AA3-lteUUKVE_CGOWB7YeBFhYLzOlDsaks5qpq0ZgaJpZM4J7nzs>.
>